src/solvers/refinement/string_refinement.cpp

```diff
@@ -1955,50 +2061,34 @@ exprt substitute_array_lists(exprt expr, size_t string_max_length)
 /// \return an expression
 exprt string_refinementt::get(const exprt &expr) const
 {
-  const std::function<exprt(const exprt &)> super_get=[this](const exprt &expr)
-  { return (exprt) supert::get(expr); };
+  const auto super_get=[this](const exprt &expr)
+  { return supert::get(expr); };
   exprt ecopy(expr);
-  replace_expr(symbol_resolve, ecopy);
-  if(is_char_array(ns, ecopy.type()))
-  {
-    auto it_content=found_content.find(ecopy);
-    if(it_content!=found_content.end())
-      return it_content->second;
+  (void)symbol_resolve.replace_expr(ecopy);
 
-    auto it=found_length.find(ecopy);
-    if(it!=found_length.end())
-      return get_array(
-        super_get,
-        ns,
-        generator.max_string_length,
-        ecopy,
-        it->second);
-  }
-  else if(ecopy.id()==ID_struct)
+  if(ecopy.type().id()==ID_array)
```

Comment:
Should this be all arrays? It used to just be character arrays. For instance, the skolem functions in not contains constraints are arrays. This code seems to assume the arrays are strings.